doseCalc/+DoseEngines/matRad_ParticlePencilBeamEngineAbstract.m

```diff
@@ -0,0 +1,838 @@
+            if this.airOffsetCorrection
+                if ~isfield(this.machine.meta, 'fitAirOffset')
+                    this.machine.meta.fitAirOffset = 0; %By default we assume that the base data was fitted to a phantom with surface at isocenter
+                    matRad_cfg.dispDebug('Asked for correction of Base Data Air Offset, but no value found. Using default value of %f mm.\n',this.machine.meta.fitAirOffset);
+                end
+            else
+                this.machine.meta.fitAirOffset = 0;
+            end
+
+            if ~isfield(this.machine.meta, 'BAMStoIsoDist')
+                this.machine.meta.BAMStoIsoDist = 1000;
+                matRad_cfg.dispWarning('Machine data does not contain BAMStoIsoDist. Using default value of %f mm.',this.machine.meta.BAMStoIsoDist);
+            end
+            
+            if this.bioParam.bioOpt
+                this.calcBioDose = true;
+            end
+
+            % Load biologicla base data if needed
+            if this.calcBioDose
+                dij = this.loadBiologicalBaseData(cst,dij);
+                % allocate alpha and beta dose container and sparse matrices in the dij struct,
+                % for more informations see corresponding method
+                dij = this.allocateBioDoseContainer(dij);
+            end           
+
+            % allocate LET containner and let sparse matrix in dij struct
+            if this.calcLET
+                if isfield(this.machine.data,'LET')
+                    dij = this.allocateLETContainer(dij);
+                else
+                    matRad_cfg = MatRad_Config.instance();
+                    matRad_cfg.dispWarning('LET not available and will not be computed!');
+                    this.calcLET = false;
+                end
+            end
+
+            % lateral cutoff for raytracing and geo calculations
+            this.effectiveLateralCutOff = this.geometricLateralCutOff;
+        end
+
+        function currBeam = initBeam(this,dij,ct,cst,stf,i)
+            currBeam = initBeam@DoseEngines.matRad_PencilBeamEngineAbstract(this,dij,ct,cst,stf,i);
+            
+            %currBeam.rot_coordsVdoseGrid = currBeam.rot_coordsVdoseGrid(~isnan(currBeam.radDepthVdoseGrid{1}),:);
+            maxEnergy = max([currBeam.ray.energy]);
+            maxEnergyIx = find(this.round2(maxEnergy,4) == this.round2([this.machine.data.energy],4));
+            raShis = [currBeam.ray.rangeShifter];
+            minRaShi = min([raShis.eqThickness]);
+            radDepthOffset = this.machine.data(maxEnergyIx).offset + minRaShi;
+
+            % apply limit in depth
+            subSelectIx = currBeam.radDepths{1} < (this.machine.data(maxEnergyIx).depths(end) - radDepthOffset);
+            currBeam.ixRadDepths = currBeam.ixRadDepths(subSelectIx);
+            currBeam.subIxVdoseGrid = currBeam.subIxVdoseGrid(subSelectIx);
+            currBeam.radDepths = cellfun(@(rd) rd(subSelectIx),currBeam.radDepths,'UniformOutput',false);
```

Comment:
I think here currBeam.geoDepths is not filtered by subSelectIx and is kept in full length.
After in `matRad_ParticlePencilBeamEngineAbstract.getRayGeometryFromBeam()`  geoDepths and radDepths are arrays with different number of elements but the same indexes `ix` are selected on the two arrays.

Reply:
Yes, there is some inconsistency between geodepths and raddepths.